source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1401,6 +1430,27 @@ void Filter::addDynamicMetadata(const ProcessorState& state, ProcessingRequest&
   auto* cb = state.callbacks();
   envoy::config::core::v3::Metadata forwarding_metadata;
 
+  // Forward cluster metadata if so configured.
+  const auto& cluster_info = cb->streamInfo().upstreamClusterInfo();
```

Comment:
```suggestion
  const auto cluster_info = cb->streamInfo().upstreamClusterInfo();
```
The `upstreamClusterInfo` returns  a value, not a reference. While the compiler will extend the lifetime of the returned value, it reduces readability of the code.

Reply:
good point.
fixed.